### ISSUE TITLE
Statement - інструкція? Судження? Виказ? Заява? Вираз?

### ISSUE THREAD

**Issue by undead404** [webdoky/content#9]
Я готовий зайнятися словотвором

**Comment by IhorHordiichuk** [comment]
Часом мова не про оператор?

**Comment by undead404** [comment]
Ні, statement частіше перекладають як "вираз" (наприклад, 1 + 1 - це statement). @AdriandeCita у себе десь переклав як "Інструкція", я вирішив винести на обговорення

**Comment by IhorHordiichuk** [comment]
Тут залижить від контексту, може бути й інструкція, може бути й вираз. Мені трапілось було "with statement". Мова йшла про застосування "with", тобто з контексту випливало, що це саме про оператор.

**Comment by AdriandeCita** [comment]
я не дуже хочу його називати виразом, бо вираз - це `expression`. Оператор підійшов би краще, якби у нас не було іще й слова `operator`.

Спроба нагуглити щось по темі привела до таких висновків:
- Operator - оператор - це якась атомарна операція, оформлена у ключове слово або знак (`with`, `+`, `=`). Здебільшого має сенс у контексті синтаксису мови
- Expression - вираз - це певний набір операторів і операндів (сюди ж входять числа і літерали). Пригадую, часто використовувалось у математиці в контексті "вираз в дужках". З цього також виходить, що вираз може включати в себе оператори
- Statement - інструкція? - це оформлена операція, яка щось робить. Виходить, що за змістом воно близько до оператора, але оператор стосується лексики, а statement - коду загалом.

Тут починається плутанина, оскільки statement можуть складатися з виразів і операторів, але й деякі вирази можуть розділятися на statements...

Знайшов іще один підхід, який розділяє їх так:
- Expression - це вираз, який обраховує якесь значення (знову ж таки, "вираз в дужках")
- Statement - це вираз, який щось робить (`with` з цієї точки зору саме statement, а не expression).

Statement `with` (як одиниця логіки) складається з оператора `with` (одиниця мови JS).

**Comment by AdriandeCita** [comment]
найкорисніше джерело https://stackoverflow.com/questions/4728073/what-is-the-difference-between-an-expression-and-a-statement-in-python

Коротше кажучи, ці три штуки різні, тому їх бажано не плутати. Як називати statement - я відкритий до пропозицій. З наявних мені поки більше подобається інструкція (навіть з урахуванням наявності терміну `instruction` - він частіше використовується в контексті мікроархітектури процесорів, тому тут я спокійний). Я радше проти виразу, бо там явна колізія з expression. Заява і судження - навряд, бо це більше асоціюється з юриспруденцією. Виказ - я його дуже рідко чую, тим паче в контексті ІТ.

**Comment by IhorHordiichuk** [comment]
Цей термін доведеться щоразу підганяти до контексту. Словники з математики та IT-термінів тому підтвердження https://e2u.org.ua/s?w=Statement&dicts=all&main_only=on&highlight=on

**Comment by AdriandeCita** [comment]
І так, і ні. Тобто так, звучить непогано, але англійською цей термін уже вживається у вірному контексті (здебільшого), тому навряд чи треба буде прям кожне місце уточнювати

**Comment by IhorHordiichuk** [comment]
Я цілком за «інструкцію», просто звертаю увагу, що треба слідкувати за контекстом.

**Comment by undead404** [comment]
Пишемо "інструкція" і закриваємо?

**Comment by AdriandeCita** [comment]
До речі, твердження... У мене виникало враження, що інструкція погано підходить для декларативностей, зокрема для контексту CSS. Виказ навіває вайби НКВС, але **твердження** такого недоліку не має.

Пропоную залишити два варіанти: усталений **інструкція**, і **твердження** для контекстів, де інструкція не дуже підходить

**Comment by undead404** [comment]
Вікіпедія: інструкція, твердження, звітність, предикат, пресреліз
Англійсько-український глосарій виробів Microsoft (2006): твердження, оператор
Microsoft Language Portal: інструкція

Так що тут два кандидати, судячи з практики вживання: твердження та інструкція
